src/api/cow/cow.spec.ts

```diff
@@ -0,0 +1,441 @@
+import { CowError } from '../../utils/common'
+import { OrderCreation, UnsignedOrder } from '../../utils/sign'
+
+enableFetchMocks()
+
+const chainId = 4 //Rinkeby
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const gpV2Contracts = require('@gnosis.pm/gp-v2-contracts')
+
+// @ts-expect-error: Let's ignore a compile error like this window.ethereum undefined
+const provider = new ethers.providers.Web3Provider(window.ethereum)
+
+const cowSdk = new CowSdk(chainId, { signer: provider.getSigner() })
+
+const HTTP_STATUS_OK = 200
+const HTTP_STATUS_NOT_FOUND = 404
+
+const ORDER_RESPONSE = {
+  sellToken: '0x6810e776880c02933d47db1b9fc05908e5386b96',
+  buyToken: '0x6810e776880c02933d47db1b9fc05908e5386b96',
+  receiver: '0x6810e776880c02933d47db1b9fc05908e5386b96',
+  sellAmount: '1234567890',
+  buyAmount: '1234567890',
+  validTo: 0,
+  appData: '0x0000000000000000000000000000000000000000000000000000000000000000',
+  feeAmount: '1234567890',
+  kind: 'buy',
+  partiallyFillable: true,
+  sellTokenBalance: 'erc20',
+  buyTokenBalance: 'erc20',
+  signingScheme: 'eip712',
+  signature:
+    '0x0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000',
+  from: '0x6810e776880c02933d47db1b9fc05908e5386b96',
+  creationTime: '2020-12-03T18:35:18.814523Z',
+  owner: '0x6810e776880c02933d47db1b9fc05908e5386b96',
+  uid: 'string',
+  availableBalance: '1234567890',
+  executedSellAmount: '1234567890',
+  executedSellAmountBeforeFees: '1234567890',
+  executedBuyAmount: '1234567890',
+  executedFeeAmount: '1234567890',
+  invalidated: true,
+  status: 'presignaturePending',
+  fullFeeAmount: '1234567890',
+}
+
+const TRADE_RESPONSE = {
+  blockNumber: 0,
+  logIndex: 0,
+  orderUid: 'string',
+  owner: '0x6810e776880c02933d47db1b9fc05908e5386b96',
+  sellToken: '0x6810e776880c02933d47db1b9fc05908e5386b96',
+  buyToken: '0x6810e776880c02933d47db1b9fc05908e5386b96',
+  sellAmount: '1234567890',
+  sellAmountBeforeFees: '1234567890',
+  buyAmount: '1234567890',
+  transactionHash: '0xd51f28edffcaaa76be4a22f6375ad289272c037f3cc072345676e88d92ced8b5',
+}
+
+const PRICE_QUOTE_RESPONSE = {
+  amount: '1234567890',
+  token: '0x6810e776880c02933d47db1b9fc05908e5386b96',
+}
+
+const PROFILE_DATA_RESPONSE = {
+  totalTrades: 1,
+  totalReferrals: 0,
+  tradeVolumeUsd: 24.22795306584268,
+  referralVolumeUsd: 0.0,
+  lastUpdated: '2022-04-11T15:51:06Z',
+}
+
+const SIGNED_ORDER_RESPONSE = {
+  signature:
+    '0x4d306ce7c770d22005bcfc00223f8d9aaa04e8a20099cc986cb9ccf60c7e876b777ceafb1e03f359ebc6d3dc84245d111a3df584212b5679cb5f9e6717b69b031b',
+  signingScheme: SigningScheme.EIP712,
+}
+
+beforeEach(() => {
+  fetchMock.resetMocks()
+})
+
+afterEach(() => {
+  jest.restoreAllMocks()
+})
+
+test('Valid: Get an order ', async () => {
+  fetchMock.mockResponseOnce(JSON.stringify(ORDER_RESPONSE), { status: HTTP_STATUS_OK })
+  const order = await cowSdk.cowApi.getOrder('string')
```

Comment:
Instead of harcoding the order ID, I suggest to use the value from the constant, like so:

```suggestion
  const order = await cowSdk.cowApi.getOrder(ORDER_RESPONSE.uid)
```